src/mopidy.js

```diff
@@ -7,9 +7,17 @@ function snakeToCamel(name) {
   );
 }
 
+// Transforms `fn` into one that executes on next tick instead of synchronously
+function makeAsync(fn) {
+  return function asyncFunction(...args) {
+    setImmediate(() => fn.apply(this, args));
+  };
+}
+
 class Mopidy extends EventEmitter {
   constructor(settings) {
     super();
+    this._reconnect = makeAsync(this._reconnect);
```

Comment:
Let's simplify things and only make `this._reconnect` async. The rest of the methods are just delegating websocket events and don't have internal listeners.

Reply:
Since we only do this for a single function now, maybe change the `_reconnect` implementation to something like this instead? That way, there's less action-at-a-distance where code on line 20 changes how a method at line 133 works.

```js
_reconnect() {
    setImmediate(() => {
       // Existing method body here
   })
}
```